delay/src/main/scala/com/ovoenergy/effect/Delay.scala

```diff
@@ -0,0 +1,37 @@
+package com.ovoenergy.effect
+
+import cats.FlatMap
+import cats.effect.Async
+import cats.syntax.applicative._
+import fs2.{Pipe, Scheduler, Stream}
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration.FiniteDuration
+import scala.language.higherKinds
+
+/**
+ * A type class that encodes the ability of some effect F
+ * to represent a delay in the processing of subsequent steps
+ */
+trait Delay[F[_]] {
+  def delay[A](duration: FiniteDuration): Pipe[F, A, A]
+  def sleep_[A](duration: FiniteDuration): Stream[F, Nothing]
+}
+
+object Delay {
+  type StreamDelay[F[_]] = Stream[F, Scheduler]
+
+  implicit def AsyncDelay[F[_]: Async: FlatMap](
```

Comment:
`FlatMap` is redundant here since it's already in `Async`.
`Async` <- `Sync` <- `MonadError` <- `Monad` <- `FlatMap`.

Reply:
yeah, I removed it from effect-utils version also